core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -78,15 +81,19 @@ public void filter(ContainerRequestContext request)
             request.abortWith(Response.seeOther(DISABLED_LOCATION_URI).build());
             return;
         }
-
-        Optional<String> subject = getAccessToken(request).map(token -> token.getBody().getSubject());
-        if (subject.isEmpty()) {
+        Optional<Claims> claims = getAccessToken(request).map(Jws::getBody);
+        if (claims.isEmpty()) {
             needAuthentication(request);
             return;
         }
+        if (!hasValidAudience(claims.get())) {
```

Comment:
Isn't this leaking secure information? It reveals that the audience is wrong. Would it be better to just act is if auth is needed?

Reply:
I suggest logging (debug) and just erroring with "Unauthorized"